src/BMPDecoder.ts

```diff
@@ -21,45 +23,123 @@ export default class BMPDecoder {
     this.pixelDataOffset = this.bufferData.skip(8).readUint32();
     this.width = this.bufferData.skip(4).readUint32();
     this.height = this.bufferData.readUint32();
-    this.bitDepth = this.bufferData.seek(28).readUint16();
+    this.bitsPerPixel = this.bufferData.seek(28).readUint16();
+    if (this.bitsPerPixel === 16 || this.bitsPerPixel === 4) {
+      throw new Error('4 and 16 bits per pixel are not supported.');
+    }
+    this.compression = this.bufferData.readUint32();
+    if (this.compression !== 0 && this.compression !== 3) {
+      throw new Error(
+        'Only BI_RGB and BI_BITFIELDS compression methods are allowed.'
+      );
+    }
+
+    this.colorMasks = [
+      this.bufferData.seek(54).readUint32(),
+      this.bufferData.readUint32(),
+      this.bufferData.readUint32(),
+    ];
+
+    if (
+      this.bitsPerPixel === 32 &&
+      (this.colorMasks[0] !== 0x00ff0000 ||
+        this.colorMasks[1] !== 0x0000ff00 ||
+        this.colorMasks[2] !== 0x000000ff)
+    ) {
+      throw new Error(
+        'These color masks are not supported by this color model.'
```

Comment:
"These" does not refer to anything when the user reads the message. I suggest you rephrase.